### PR TITLE
Refresh on save, remove duplicate edge in juggl

### PR DESCRIPTION
**Pull request by HananoshikaYomaru**
this pull request solve two issues. 

# repeated edge in juggl codeblock

As hiererchy note will make real up and real down, juggl will show two edges overlapping with each other.

Add a setting `showUpInJuggl : boolean`. When this is false, no more repeated edge, only down is shown.

# no refresh on save

add a setting `refreshOnNoteSave : boolean`, when this is check, refresh index on save🤙🏻

# bug on adjust hierarchy manipulator when using hierarchy note #330 

Hierarchy note show nothing when using hierarchy note folder. The bug is fixed. 

Also, `enter / click` is changed to `shift + enter` because the purpose of the quick adjust panel is to adjust without going into the hierarchy notes. Therefore, the most common key should map to `add child`. 

Also, the default value should not be `<empty>`. It should be the file name of active leaf. 

# TODO 

- [ ] #327

### CODE REVIEW

File: src/AlternativeHierarchies/HierarchyNotes/HierarchyNotes.ts [hunk 1]
```diff
@@ -22,15 +22,16 @@ export async function getHierarchyNoteItems(plugin: BCPlugin, file: TFile) {
   const problemFields: string[] = [];
 
   const upFields = getFields(userHiers, "up");
+  const nextFields = getFields( userHiers , "next")
```
Comment (HEmile) [inline src/AlternativeHierarchies/HierarchyNotes/HierarchyNotes.ts:25]: What are those nextFields for?

Comment (HEmile) [reply src/AlternativeHierarchies/HierarchyNotes/HierarchyNotes.ts:25]: It seems like you're also changing something in hierarchy notes with there being next and previous links which is not explained in the PR.

File: src/Settings/GeneralSettings.ts [hunk 2]
```diff
@@ -74,6 +74,16 @@ export function addGeneralSettings(plugin: BCPlugin, containerEl: HTMLElement) {
       })
     );
 
+  new Setting (generalDetails) 
+  .setName("show up fields in Juggl")
```
Comment (HEmile) [inline src/Settings/GeneralSettings.ts:87]: Capitalize name. Also, make this a bit more clear in the description why this is needed. @SkepticMystic is this a good place to put this setting?

File: src/Settings/MatrixViewSettings.ts [hunk 3]
```diff
@@ -188,5 +188,6 @@ export function addMatrixViewSettings(
         await plugin.saveSettings();
         await plugin.getActiveTYPEView(MATRIX_VIEW).draw();
       })
-    }
+    })
```
Comment (HEmile) [inline src/Settings/MatrixViewSettings.ts:191]: This change means it either introduces a syntax error, or there was already a syntax error... right?

File: src/interfaces.ts [hunk 4]
```diff
@@ -96,6 +96,7 @@ export interface BCSettings {
   showAll: boolean;
   showGrid: boolean;
   showImpliedRelations: boolean;
+  showUpInJuggl : boolean  ;
```
Comment (HEmile) [inline src/interfaces.ts:100]: I'm not seeing a default value? Should probably be false.

**Review by HEmile — COMMENTED** [review]
I'd recommend splitting up the two commits as they are introducing very different features. What is the motivation for the second commit? @SkepticMystic Maybe the second commit can better be merged as replacing the on-change event as it might be more efficient?

File: src/Settings/MatrixViewSettings.ts (hunk 3 above)
Comment (HananoshikaYomaru) [reply src/Settings/MatrixViewSettings.ts:191]: yes it is a syntax error

File: src/interfaces.ts (hunk 4 above)
Comment (HananoshikaYomaru) [reply src/interfaces.ts:100]: yes it is false originally.

File: src/AlternativeHierarchies/HierarchyNotes/HierarchyNotes.ts (hunk 1 above)
Comment (HananoshikaYomaru) [reply src/AlternativeHierarchies/HierarchyNotes/HierarchyNotes.ts:25]: oops. I forget remove this. I was trying to implement the feature I was explaining in #327. But it is useless in this two context

**Comment by HananoshikaYomaru** [comment]
@SkepticMystic @HEmile I also fix the issue on adjust hierarchy manipulator. Please check.

**Comment by SkepticMystic** [comment]
@HananoshikaYomaru the way I ended up merging means the changes won't be credited to you on the Contributors page. I can try do it again to make it show that you made these changes?
But I've already adjusted some of your code, so my changes would be deleted.
Let me know what you think

**Comment by HananoshikaYomaru** [comment]
Thanks for testing for me. I am very happy to contribute to breadcrumbs. I definitely want my contribution show up in the repos. So it might be a bit troublesome but please help me adjust it back.

**Comment by SkepticMystic** [comment]
Cool cool, I think this worked

**Pull request merged**